machines/get-connection.js

```diff
@@ -148,24 +167,44 @@ module.exports = {
       client.on('error', function onIntraConnectionError (err){
         // If manager was not provisioned with an `onUnexpectedFailure`,
         // we'll just handle this error event silently (to prevent crashing).
-        if (!isFunction(inputs.manager.onUnexpectedFailure)) {
+        if (!_.isFunction(inputs.manager.onUnexpectedFailure)) {
           return;
         }
 
+        var errToSend = new Error;
+        errToSend.connection = client;
```

Comment:
note for future reference, this can be accomplished w/ flaverr by doing `flaverr({ connection: client, failureType: 'error' }, new Error('...'))`